packages/firestore/test/unit/specs/write_spec.test.ts

```diff
@@ -996,4 +996,98 @@ describeSpec('Writes:', [], () => {
         .writeAcks('collection/b', 2000);
     }
   );
+
+  specTest('Secondary tabs handle user change', ['multi-client'], () => {
+    const query = Query.atPath(path('collection'));
+    const docALocal = doc(
+      'collection/a',
+      0,
+      { v: 1 },
+      { hasLocalMutations: true }
+    );
+    const docBLocal = doc(
+      'collection/b',
+      0,
+      { v: 1 },
+      { hasLocalMutations: true }
+    );
+    const docCLocal = doc(
+      'collection/c',
+      0,
+      { v: 1 },
+      { hasLocalMutations: true }
+    );
+
+    // Firebase Auth attempts to rapidly synchronize user changes across tab. We
+    // emulate this behavior in this spec tests by calling `changeUser` manually
+    // for all clients.
```

Comment:
FWIW, I think we should probably tolerate if auth doesn't successfully do this (or there's a delay)...  I'm tempted to ask you to write a test or two that intentionally has different tabs operating as different users and makes sure we still end up with sensible well-defined behavior.  But you could also try to convince me that it will "just work" or that we shouldn't really care (because we can trust auth to synchronize?)...  WDYT?

Reply:
It took me 10 minutes to write this test and then about 2 hours to figure out why it wasn't passing. I was changing the user at the wrong time, and my mutations didn't get synced properly. If a user changes happens in tab A, but hasn't happened in tab B, then tab B will ignore the mutations for the new user and pick them up as we reload the mutation queue. Tab A will also ignore the mutations from the old user (this is all done in SharedUserState).

Reply:
... oh and I also just changed my mind on this and added a test.